### PR TITLE
Declarative modules: make sure to emit doc comments and other attributes

### PR DESCRIPTION
**Pull request by Tpt**
Reported in https://github.com/PyO3/pyo3/issues/3900#issuecomment-2151679160

### CODE REVIEW

**Comment by alex** [comment]
Given the existing tests miss this, can you add a new test?

**Comment by Tpt** [comment]
> Given the existing tests miss this, can you add a new test?

Not sure how to write it without having to write a proc macro. I don't think it's possible to selectively enable the `missing_doc` lint. Do you have an idea?

**Comment by Icxolu** [comment]
Can't you just add it to the `pymodule_missing_docs` ui test?

**Comment by Tpt** [comment]
> Can't you just add it to the `pymodule_missing_docs` ui test?

Indeed. Done. Thank you!

File: tests/ui/pymodule_missing_docs.rs [hunk 1]
```diff
@@ -9,4 +9,9 @@ pub fn python_module(_m: &Bound<'_, PyModule>) -> PyResult<()> {
     Ok(())
 }
 
+#[cfg(feature = "experimental-declarative-modules")]
+/// Some module documentation
+#[pymodule]
+pub mod python_module {}
```
Comment (davidhewitt) [inline tests/ui/pymodule_missing_docs.rs:15]:
```suggestion
pub mod declarative_python_module {}
```

**Review by Icxolu — APPROVED** [review]

**Pull request merged**